registration/api/v2/views.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Registration api views for API v2
+"""
+import logging
+
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.mixins import CreateModelMixin, ListModelMixin, RetrieveModelMixin, UpdateModelMixin
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+from simple_email_confirmation.models import EmailAddress
+
+from instance.models.appserver import Status
+from instance.tasks import spawn_appserver
+from opencraft.swagger import VALIDATION_AND_AUTH_RESPONSES, VALIDATION_RESPONSE, viewset_swagger_helper
+from registration.api.v2.serializers import AccountSerializer, OpenEdXInstanceConfigSerializer
+from registration.models import BetaTestApplication
+from registration.utils import verify_user_emails
+from userprofile.models import UserProfile
+
+logger = logging.getLogger(__name__)
+
+
+@viewset_swagger_helper(
+    create="Create new user registration",
+    list="Get current user registration data",
```

Comment:
I think it's sufficient that the list method returns only the current user's profile. That can be used by the frontend without any need for a /current/ model since these APIs are only for regular users, not admins.

Reply:
Sure, that makes sense too.